### PR TITLE
Update to allow empty schema

### LINKED ISSUE #8174: Missing unit tests for Lisk Codec regarding empty Lisk JSON schema
There are no unit tests for Lisk Codec where the following schema is used:

```
emptyLiskJSONSchema = {
    "type": "object",
    "required": [],
    "properties": {}
}
```

As this schema poses an edge case, the following tests should be added:

1. `encode(emptyLiskJSONSchema, { })`. Expected return value: empty buffer.
2. `decdoe(emptyLiskJSONSchema, empty buffer)`. Expected return value: `{ }`.
3. `decdoe(emptyLiskJSONSchema, non-empty buffer)`. Expectation: decoding fails.

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #8174 

### How was it solved?

- Update schema validation to allow empty schema

### How was it tested?

- Add tests to check the empty schema behaviors

### CODE REVIEW

File: elements/lisk-validator/test/lisk_validator.spec.ts [hunk 1]
```diff
@@ -109,12 +108,11 @@ describe('validator', () => {
 			expect(() => validator.validateSchema(invalidSchema)).toThrow(msg);
 		});
 
-		it('should return error when "properties" are empty object', () => {
+		it('should not return error when "properties" are empty object', () => {
 			const invalidSchema = cloneDeep(validSchema);
```
Comment (AndreasKendziorra) [inline elements/lisk-validator/test/lisk_validator.spec.ts:112]: It would be not call it "invalidSchema" anymore.

**Review by AndreasKendziorra — CHANGES_REQUESTED** [review]

**Review by AndreasKendziorra — APPROVED** [review]

**Review by bobanm — APPROVED** [review]
